### ISSUE TITLE
Events not sent due to deepMerge crash on object with circular references

### ISSUE THREAD

**Issue by ilovett** [DataDog/browser-sdk#587]
```
utils.js?3d73:86 Uncaught RangeError: Maximum call stack size exceeded
    at deepMerge (utils.js?3d73:86)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
    at deepMerge (utils.js?3d73:118)
```

After some debugging found that one of the things I was logging had an object with a circular reference.

I think the error might somehow be able to be caught and give this hint in an error message, or possibly even avoid the circular reference some how would be better.

Not sure how or why it wasn't caught in previous versions, It did not happen in 1.19.0

**Comment by bcaudan** [comment]
Hello @ilovett,

Thanks for reporting this.
I've tried to reproduce with:

``` js
function Foo() {
  this.circular = this;
}
var foo = new Foo();

DD_LOGS.logger.log(foo)
```

the log is sent as:
```
RangeError: Maximum call stack size exceeded
  at v @ https://www.datadoghq-browser-agent.com/datadog-logs.js:1:3135
  at v @ https://www.datadoghq-browser-agent.com/datadog-logs.js:1:3391
  at v @ https://www.datadoghq-browser-agent.com/datadog-logs.js:1:3391
  at v @ https://www.datadoghq-browser-agent.com/datadog-logs.js:1:3391
...
``` 

Is it the behavior you observed?
or do you have an infinite loop that freeze the UI?

**Comment by ilovett** [comment]
@bcaudan Yes, I got a bunch of other obscure errors caused by the first one being the `deepMerge`.  It did not freeze the UI, because it seems the internal error prevented the freeze due to maximum calls stack depth.

**Comment by bcaudan** [comment]
OK, thanks, we will have a look at it!

**Comment by bcaudan** [comment]
@ilovett this should be fixed in 1.25.1
